src/app/components/report/report.component.ts

```diff
@@ -0,0 +1,41 @@
+import { CommonModule } from '@angular/common';
+import { Component, OnInit } from '@angular/core';
+import { MatCardModule } from '@angular/material/card';
+import { MatIconModule } from '@angular/material/icon';
+import { ReportsService } from '../../services/reports/reports.service';
+import { ActivatedRoute, Params } from '@angular/router';
+import { Report } from '../../models/report';
+import { HttpResponse } from '@angular/common/http';
+import { ResponseType } from '../../models/response-type';
+
+@Component({
+  selector: 'app-report',
+  standalone: true,
+  imports: [
+    MatCardModule, 
+    CommonModule,
+    MatIconModule
+  ],
+  templateUrl: './report.component.html',
+  styleUrl: './report.component.scss'
+})
+export class ReportComponent implements OnInit{
+  id: string = '';
+  report: Report | undefined;
+  constructor(private reportsService: ReportsService,
+              private route: ActivatedRoute) {
+
+  this.route.params.subscribe(
+    (params: Params) => {
+      this.id = params['id'];
+    });
+  
+  }
+
+  ngOnInit(): void {
+    this.reportsService.getReportById(this.id).subscribe((response: HttpResponse<ResponseType<Report>>) => {
```

Comment:
@SelmaAhmetovic , zasto nisi koristila RXJS metode kao sto je map? Da li je uradjen unsubscribe kako bi se sprijecilo curenje memorije?

Teoretski: sta ako se pocne izvrasavati this.blogsService.getBlogById a ti jos uvijek nemas vrijednost za "this.id" ?